apps/shared/avifpng.c

```diff
@@ -347,6 +423,17 @@ avifBool avifPNGRead(const char * inputFilename,
         rowPointers[y] = &rgb.pixels[y * rgb.rowBytes];
     }
     png_read_image(png, rowPointers);
+    // gAMA chunk should be applied if it has an accompanying cHRM chunk, AND there is no overriding sRGB or iCCP chunk.
+    if (!ignoreICC && hasGama && hasChrm && !hasSrgb && !hasICC) {
```

Comment:
I checked the Chrome source code in third_party/blink/renderer/platform/image-decoders/png/png_image_decoder.cc. It seems that Chrome also builds an ICC profile (in decoded form) from cHRM and gAMA. Is this correct?